x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/rate/AbstractRateAggregator.java

```diff
@@ -46,7 +40,7 @@ public RateAggregator(
         Map<String, Object> metadata
     ) throws IOException {
         super(name, context, parent, metadata);
-        this.valuesSource = (ValuesSource.Numeric) valuesSourceConfig.getValuesSource();
+        this.valuesSource = valuesSourceConfig.getValuesSource();
```

Comment:
I think this is fine, but just for the sake of mentioning options, we could push getting the values source into the concrete subclasses and do the cast in the constructor, rather than at access time in the leaf reader.